File: apple-and-orange.java
class Result {
    public static void countApplesAndOranges(int s, int t, int a, int b, List<Integer> apples, List<Integer> oranges) {
        int count_a = 0;
        int count_b = 0;
        for(int val : apples){
            int pos = a+val;
            if(pos >= s && pos <= t){
                count_a++;
            }
        }
        for(int val : oranges){
            int pos = b+val;
            if(pos >= s && pos <= t){
                count_b++;
            }
        }
        System.out.println(count_a);
        System.out.println(count_b);
    }
}

File: apple-and-orange.py
def countApplesAndOranges(s, t, a, b, apples, oranges):
    count_a = count_o = 0
    
    for apple in apples:
        pos = a + apple
        if pos>=s and pos <=t:
            count_a+=1
    print(count_a)
            
    for orange in oranges:
        pos = b + orange
        if pos>=s and pos <=t:
            count_o+=1
    print(count_o)